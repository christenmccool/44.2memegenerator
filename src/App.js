import './App.css';
import MemeForm from "./MemeForm";
import MemeList from "./MemeList";

import {useSelector, useDispatch} from 'react-redux';


function App() {
  const memes = useSelector(store => store.memes);
  const dispatch = useDispatch();

  const addMeme = (meme) => {
    dispatch({ type: "ADD_MEME", payload: meme });
  };
  
  const removeMeme = (id) => {
    dispatch({type:"REMOVE_MEME", id});
  };

  return (
    <div className="App">
      <h1>Meme Generator</h1>
      <MemeForm addMeme={addMeme}/>
      <MemeList memes={memes} removeMeme={removeMeme}/>
    </div>
  );
}

export default App;
