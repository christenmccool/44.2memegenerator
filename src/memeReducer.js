const memesFromStorage = JSON.parse(localStorage.getItem('memes')) || [];

// const INITIAL_STATE = {memes: 
//                         [
//                           {id: 1,
//                             url:"https://www.cheatsheet.com/wp-content/uploads/2020/05/Steve-Carell.jpg",                          
//                             top:"No God!",
//                             bottom: "No please no!"},
//                           {id: 2,
//                             url:"https://i.redd.it/4nodbxs79u311.jpg",
//                             top:"No God!",
//                             bottom: "No please no!"}
//                         ]
//                       };

const INITIAL_STATE = {memes: memesFromStorage};

const memeReducer = (state=INITIAL_STATE, action) => {
  switch (action.type) {
    case "ADD_MEME": {
      localStorage.setItem('memes', JSON.stringify([...state.memes, action.payload]));
      return {...state, 
              memes: [...state.memes, action.payload]};
      }
    case "REMOVE_MEME": {
      localStorage.setItem('memes', JSON.stringify(state.memes.filter(ele => ele.id !== action.id)));
      return {...state, 
              memes: state.memes.filter(ele => ele.id !== action.id)};
      }
    default: return state;
  }
}

export default memeReducer;