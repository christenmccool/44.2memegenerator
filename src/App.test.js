import { render, asFragment } from '@testing-library/react';
import {createStore} from "redux";
import {Provider} from "react-redux";
import memeReducer from './memeReducer';
import App from './App';

const store = createStore(memeReducer);

test('renders without crashing', () => {
  render(<Provider store={store}><App /></Provider>);
  }
);

test('matches snapshot', () => {
  const {asFragment} = render(<Provider store={store}><App /></Provider>);
  expect(asFragment()).toMatchSnapshot();
  }
);

test('shows title and form', () => {
  const {queryByText} = render(<Provider store={store}><App /></Provider>);
  expect(queryByText('Meme Generator')).toBeInTheDocument();
  expect(queryByText('Url:')).toBeInTheDocument();
  expect(queryByText('Top:')).toBeInTheDocument();
  expect(queryByText('Bottom:')).toBeInTheDocument();
  expect(queryByText('Make meme')).toBeInTheDocument();
  }
);

