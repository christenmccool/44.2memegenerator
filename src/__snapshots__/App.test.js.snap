// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Meme Generator
    </h1>
    <div
      class="MemeForm"
    >
      <form
        class="MemeForm-form"
      >
        <div
          class="MemeForm-group"
        >
          <label
            class="MemeForm-label"
            for="url"
          >
            Url:
          </label>
          <input
            class="MemeForm-input"
            id="url"
            name="url"
            value=""
          />
        </div>
        <div
          class="MemeForm-group"
        >
          <label
            class="MemeForm-label"
            for="Top"
          >
            Top:
          </label>
          <input
            class="MemeForm-input"
            id="top"
            name="top"
            value=""
          />
        </div>
        <div
          class="MemeForm-group"
        >
          <label
            class="MemeForm-label"
            for="bottom"
          >
            Bottom:
          </label>
          <input
            class="MemeForm-input"
            id="bottom"
            name="bottom"
            value=""
          />
        </div>
        <div
          class="MemeForm-group"
        >
          <button
            class="MemeForm-button"
            type="submit"
          >
            Make meme
          </button>
        </div>
      </form>
    </div>
    <div
      class="MemeList"
    />
  </div>
</DocumentFragment>
`;
